Description: Commit 94560ab57d9fc23673f42017e6f2a78cb2b66b7a added a fast-path
 to avoid work when baremetal compute nodes haven't changed, but if
 the compute message has None for its updated-at timestamp (which
 can happen during startup) the comparison crashes.
Author: Robert Collins <robertc@robertcollins.net>
Origin: upstream
https://launchpad.net/bugs/1096719
Forwarded: not-needed
Last-Update: 2013-03-21

---
 nova/scheduler/host_manager.py |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/nova/scheduler/host_manager.py b/nova/scheduler/host_manager.py
index 8f43954..e406731 100644
--- a/nova/scheduler/host_manager.py
+++ b/nova/scheduler/host_manager.py
@@ -122,7 +122,8 @@ class HostState(object):
 
     def update_from_compute_node(self, compute):
         """Update information about a host from its compute_node info."""
-        if self.updated and self.updated > compute['updated_at']:
+        if (self.updated and compute['updated_at']
+            and self.updated > compute['updated_at']):
             return
         all_ram_mb = compute['memory_mb']
 
-- 
1.7.10.4

